resource/utilities/rq2.cpp

```diff
@@ -69,16 +225,149 @@ int info (resource_query_t &ctx)
     return rc;
 }
 
+static void process_args (json_t *options,
+                          int argc, char *argv[])
+{
+    int rc = 0;
+    int ch = 0;
+    std::string token;
+
+    while ((ch = getopt_long (argc, argv, OPTIONS, longopts, NULL)) != -1) {
+        switch (ch) {
+            case 'h': /* --help */
+                usage (0);
+                break;
+            case 'L': /* --load-file */
+                json_object_set_new(options, "load_file", json_string(optarg));
+                if (!fs::exists(optarg)) {
+                    std::cerr << "[ERROR] file does not exist for --load-file: ";
+                    std::cerr << optarg << std::endl;
+                    usage (1);
+                } else if (fs::is_directory(json_string_value(json_object_get(options, "load_file")))) {
+                    std::cerr << "[ERROR] path passed to --load-file is a directory: ";
+                    std::cerr << optarg << std::endl;
+                    usage (1);
```

Comment:
I think printing the usage message (which is huge) on every error is too much. The error message will get lost. Maybe print the error message and then another message saying "Use `-h` for usage information"?

Reply:
That's a good point. This particular block of code is copied directly from resource-query, so I may be concerned that changing the expected outputs might make some tests fail? It may be that none of the tests specifically test the argument parsing though